Sources/StreamChat/Models/MessageModerationDetails.swift

```diff
@@ -10,20 +10,28 @@ public struct MessageModerationDetails {
     public let originalText: String
     /// The type of moderation performed to a message.
     public let action: MessageModerationAction
+    /// Array of harm labels found in text.
+    public let textHarms: [String]?
+    /// Array of harm labels found in images.
+    public let imageHarms: [String]?
+    /// Blocklist name that was matched.
+    public let blocklistMatched: [String]?
+    /// Semantic filter phrase list that was matched.
+    public let semanticFilterMatched: [String]?
+    /// A boolean value indicating if the message triggered the platform circumvention model.
+    public let platformCircumvented: Bool?
 }
 
 /// The type of moderation performed to a message.
-public struct MessageModerationAction: RawRepresentable, Equatable {
-    public let rawValue: String
+public struct MessageModerationAction: Equatable {
+    let rawValue: String
 
-    public init(rawValue: String) {
+    internal init(rawValue: String) {
         self.rawValue = rawValue
     }
 
-    /// A bounced message means it needs to be rephrased and sent again.
-    static let bounce = Self(rawValue: "MESSAGE_RESPONSE_ACTION_BOUNCE")
-    /// A flagged message means it was sent for review in the dashboard but the message was still published.
-    static let flag = Self(rawValue: "MESSAGE_RESPONSE_ACTION_FLAG")
-    /// A blocked message means it was not published and it was sent for review in the dashboard.
-    static let block = Self(rawValue: "MESSAGE_RESPONSE_ACTION_BLOCK")
+    /// The message was bounced message, which means it needs to be rephrased and sent again.
+    public static let bounce = Self(rawValue: "bounce")
+    /// The message was blocked and removed from the chat.
+    public static let remove = Self(rawValue: "remove")
```

Comment:
No problem to change this because the previous actions were mistakenly private. So customers could not access the type of the action before.

Reply:
Hum, probably better to make it public yes, I will change this

Reply:
I probably removed RawRepresentable because it required rawValue to be public, as well as the initialiser. But I guess it is better if we make it public to have more flexibility.

Reply:
Done ✅